src/cider/nrepl/middleware/classpath.clj

```diff
@@ -2,10 +2,17 @@
   (:require
    [cider.nrepl.middleware.util.error-handling :refer [with-safe-transport]]
    [clojure.java.io :as io]
-   [orchard.java.classpath :as cp]))
+   [orchard.java.classpath :as cp]
+   [orchard.misc :as misc]))
+
+(defn file-url?
+  [u]
+  (and (misc/url? u)
+       (= (.getProtocol ^java.net.URL u) "file")))
 
 (defn classpath-reply [msg]
   {:classpath (->> (cp/classpath)
```

Comment:
I've also been thinking we can just add a `classpath-files` function to orchard.

Reply:
I wonder if that would filter out too much for other clients of orchard. Maybe some clients want to fix this problem instead of just removing the classpath entries which causes this function to fail. Would it not take that possibility from them if we filter on the orchard side?

Reply:
Well, adding the function there wouldn't mean that people are forced to use it. It simply seems to me like something that's generally useful outside `cider-nrepl`.